client/X11/xf_disp.c

```diff
@@ -341,11 +383,16 @@ UINT xf_DisplayControlCaps(DispClientContext* disp, UINT32 maxNumMonitors,
 	return xf_disp_set_window_resizable(xfDisp) ? CHANNEL_RC_OK : CHANNEL_RC_NO_MEMORY;
 }
 
-BOOL xf_disp_init(xfContext* xfc, DispClientContext* disp)
+BOOL xf_disp_init(xfDispContext* xfDisp, DispClientContext* disp)
 {
-	rdpSettings* settings = xfc->context.settings;
-	xfc->disp = disp;
-	disp->custom = (void*) xfc->xfDisp;
+	rdpSettings* settings;
+
+	if (!xfDisp || !xfDisp->xfc || !disp)
+		return FALSE;
+
```

Comment:
I don't understand the rational for these checks, I can't see any situation that would lead any of these to be NULL.

Reply:
@hardening I agree with some of these checks but especially in this function which is called from a handler and therefore possibly externally invoked. It makes sense imho.